src/renderer/ducks/validator/sagas.ts

```diff
@@ -335,6 +341,87 @@ function* onBeaconNodeStatusChangeUpdateValidatorState(
     }
 }
 
+export function* getAttestationEffectiveness({
+    payload,
+    meta,
+}: ReturnType<typeof signedNewAttestation>): Generator<
+    SelectEffect | TakeEffect | AllEffect<CallEffect> | CallEffect,
+    void,
+    IValidatorComplete & ReturnType<typeof updateSlot> & (BlockAttestations[] | null)[]
+> {
+    const validator = yield select(getValidator, {publicKey: meta});
+    const config = getNetworkConfig(validator.network)?.eth2Config || mainnetConfig;
+
+    const eth2API = new CgEth2ApiClient(config, validator.beaconNodes[0]);
+
+    let lastSlot = payload.slot,
+        previousSlot = 0,
+        empty = 0,
+        skipped = 0,
+        skippedQue = 0,
+        inclusion = 0;
+    const timeoutSlot = config.params.SLOTS_PER_EPOCH + payload.slot;
+    while (true) {
+        const newSlot = yield take(updateSlot);
+        if (newSlot.meta !== validator.beaconNodes[0]) continue;
+        if (lastSlot >= newSlot.payload) continue;
+
+        // get all block information from last to current (mostly is only one but is important to get every block)
+        const range = new Array(newSlot.payload - lastSlot).fill(null).map((_, index) => lastSlot + index + 1);
+        const results = yield all(range.map((slot) => call(eth2API.beacon.blocks.getBlockAttestations, slot)));
+
+        results.forEach((result, index) => {
+            /** the main logic for collecting when is attestation may be included and the number of skipped blocks */
+            if (result === null) {
+                skippedQue++;
+            } else {
+                const sanitizedAttestations = result.filter(
+                    ({data}) => payload.block === data.beaconBlockRoot && data.index === payload.index,
```

Comment:
is BeaconBlockRoot Buffer? If yes, this is not a good way of equality. Use config.types.Root,equals